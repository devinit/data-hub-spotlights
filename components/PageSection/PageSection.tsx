import React, { FunctionComponent } from 'react';

const PageSection: FunctionComponent = ({ children }) => {
  return (
    <section className="section">
      <div className="row row--narrow">
        { children }
      </div>
    </section>
  );
};

export { PageSection };
