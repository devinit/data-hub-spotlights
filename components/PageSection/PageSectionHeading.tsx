import React, { FunctionComponent } from 'react';

const PageSectionHeading: FunctionComponent = ({ children }) => {
  return <h2 className="section__heading">{children}</h2>;
};

export { PageSectionHeading };
