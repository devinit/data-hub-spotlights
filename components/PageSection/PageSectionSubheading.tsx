import React, { FunctionComponent } from 'react';

const PageSectionSubHeading: FunctionComponent = ({ children }) => {
  return <h3 className="section__sub-heading">{children}</h3>;
};

export { PageSectionSubHeading };
