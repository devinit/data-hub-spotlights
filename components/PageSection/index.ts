export * from './PageSection';
export * from './PageSectionHeading';
