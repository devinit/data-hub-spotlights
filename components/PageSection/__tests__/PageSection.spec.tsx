/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { PageSection } from '../PageSection';

describe('PageSection', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<PageSection />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders the appropriate class when narrow', () => {
    const renderer = TestRenderer.create(<PageSection narrow />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders the appropriate class when wide', () => {
    const renderer = TestRenderer.create(<PageSection wide />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
