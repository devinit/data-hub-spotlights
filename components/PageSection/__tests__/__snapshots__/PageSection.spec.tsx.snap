// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSection renders correctly 1`] = `
<section
  className="section"
>
  <div
    className="row"
  />
</section>
`;

exports[`PageSection renders the appropriate class when narrow 1`] = `
<section
  className="section"
>
  <div
    className="row row--narrow"
  />
</section>
`;

exports[`PageSection renders the appropriate class when wide 1`] = `
<section
  className="section"
>
  <div
    className="row row--wide"
  />
</section>
`;
